### ISSUE TITLE
False positive for Naming/PredicateName

### ISSUE THREAD

**Issue by jcoyne** [rubocop/rubocop#10863]
I've created a method called `has_contributors` which is an assertion on a model that it has contributors (kind of like `has_many :contributors` in Rails).  This is an assertion of a property, not an interrogative, so it shouldn't recommend adding a question mark.
  
```
app/forms/contributor_form.rb:7:9: C: Naming/PredicateName: Rename has_contributors to has_contributors?.
    def has_contributors(validate:)
        ^^^^^^^^^^^^^^^^
```        

What's more is that I have this directive set, so it should only trip on "is_*":
```yaml
Naming/PredicateName:
  ForbiddenPrefixes: is_
```

```
rubocop -v
1.32.0
```

**Comment by dvandersluis** [comment]
`ForbiddenPrefixes` can't be used on its own like this. You probably want to override `NamePrefix` instead, or better still, add

```yaml
AllowedMethods:
  - has_contributors
```

**Comment by jcoyne** [comment]
@dvandersluis how is `ForbiddenPrefixes` supposed to be used? Is there documentation about it?

**Comment by dvandersluis** [comment]
No it's not documented and probably not meant to be used by end-users (I didn't even know it existed until today, and it probably should be refactored out or made internal). `AllowedMethods` is definitely the best way to handle what you're trying to do, but `NamePrefix` defines the prefixes the cop looks for.

**Comment by jcoyne** [comment]
I found this here: https://docs.rubocop.org/rubocop/cops_naming.html#configurable-attributes-12 but it didn't really say what each configuration is for, so I guess I picked the wrong value.

I still think that `has_` is a poor default as using `has_` doesn't imply an interrogative.

**Comment by dvandersluis** [comment]
Yes, when the guides are generated, all the default configuration values are enumerated, but sometimes they are not well explained. I will try to update the documentation for this cop to be more clear.

`has` and `is` have the same semantics here IMO (`is this a foo?` / `does this have foo?`) so I don't think the default should change. You certainly can make your own determination for your code base though!

**Comment by dvandersluis** [comment]
FYI I opened #10866 to improve the documentation.
